checkov/terraform/parser.py

```diff
@@ -252,7 +252,10 @@ def _process_vars_and_locals(self, directory: str,
             if not file_locals:
                 continue
             for k, v in file_locals[0].items():
-                locals_values[k] = v[0]
+                if type(v) in (float, int):
+                    locals_values[k] = v
```

Comment:
@robeden i have debugged this a little and found out this is the right line for strings too. 
when would we need `v[0]` ?

Reply:
The example is assigning a value to locals (has an equal), which is a little odd. Not sure if that’s what you’re doing, but I think that’s throwing it off. The main test for this would be the “local_block” parser scenario, though I see it’s lacking in non-string test cases.